channeldb/db.go

```diff
@@ -206,15 +206,37 @@ var (
 type DB struct {
 	kvdb.Backend
 
-	// LinkNodeDB separates all DB operations on LinkNodes.
-	LinkNodeDB
+	// ChannelStateDB separates all DB operations on channel state.
+	ChannelStateDB
 
 	dbPath string
 	graph  *ChannelGraph
 	clock  clock.Clock
 	dryRun bool
 }
 
+type ChannelStateDB struct {
+	// LinkNodeDB separates all DB operations on LinkNodes.
+	LinkNodeDB
+
+	// parent holds a pointer to the "main" channeldb.DB object. This is
+	// only used for testing and should never be used in production code.
+	// For testing use the ChannelStateDB.GetParentDB() function to retrieve
+	// this pointer.
+	parent *DB
+
+	// db points to the actual backend holding the channel state database.
+	// This may be a real backend or a cache middleware.
+	db kvdb.Backend
```

Comment:
I think we don't need this `db` as it's already in `LinkNodeDB.db`?

Reply:
The idea is to give the `LinkNodeDB` its own instance so we don't share the same reference necessarily. At the moment it will be the same instance but in the future we might point the `LinkNodeDB` to its own separate namespace.

Reply:
I decided to take things apart even more clearly in the latest state. Let me know what you think.